lib/MongoDB/ClientSession.pm

```diff
@@ -98,6 +114,34 @@ has _server_session => (
     clearer => '__clear_server_session',
 );
 
+has _current_transaction_settings => (
+    is => 'rwp',
+    isa => HashRef,
+    init_arg => undef,
+    clearer => '_clear_current_transaction_settings',
+);
+
+has _transaction_state => (
+    is => 'rwp',
+    isa => TransactionState,
+    default => 'none',
+);
+
+# Flag used to say we are still in a transaction
+has _active_transaction => (
```

Comment:
The problem is that theres no easy logic for whether we are in an active transaction - the state must be changed for committed or aborted before the command is sent (so that on error we are in the correct state), but if we are in one of those states and not in a transaction we need to reset the session state back to none - which would mean needing a flag notifying if we are in an actual commit or abort command - and the active transaction flag seems a lot more useful and can be set correctly in the right spots. Also means that setting the transaction state back to 'none' is an easy logic check!

Reply:
Ok.  Keep it, but I'll look closer after the GA release and see what can be done about it.